### PR TITLE
[Merged by Bors] - feat: define `Fin2.rev` analogously to `Fin.rev`

### PR DESCRIPTION
**Pull request by alexkeizer**
This defines `Fin2.rev`, the mapping from `i : Fin2` to (morally) `last - i`, and proves it's an involution.

---

This is code ported from https://github.com/alexkeizer/QpfTypes, and is used to revert a `TypeVec`.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by BoltonBailey — APPROVED** [review]
These all look good to me

File: Mathlib/Data/Fin/Fin2.lean [hunk 1]
```diff
@@ -127,6 +128,32 @@ def ofNat' : ∀ {n} (m) [IsLT m n], Fin2 n
...
+
+/-- Maps `0` to `n-1`, `1` to `n-2`, ..., `n-1` to `0`. -/
+def rev {n : Nat} : Fin2 n → Fin2 n
+  | .fz   => last
+  | .fs i => i.rev.castSucc
+
+@[simp] lemma rev_last {n} : rev (@last n) = fz := by
+  induction n <;> simp_all [rev, castSucc, last]
+
+@[simp] lemma inv_castSucc {n} (i : Fin2 n) : rev (castSucc i) = fs (rev i) := by
```
Comment (YaelDillies) [inline Mathlib/Data/Fin/Fin2.lean:149]:
```suggestion
@[simp] lemma rev_castSucc {n} (i : Fin2 n) : rev (castSucc i) = fs (rev i) := by
```

**Review by YaelDillies — APPROVED** [review]
Sorry for the delay!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**